src/nfd_portal.cpp

```diff
@@ -14,8 +14,17 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/random.h>  // for the random token string
-#include <unistd.h>      // for access()
+#include <unistd.h>  // for access()
+
+#if !defined(__has_include) || !defined(__linux__)
```

Comment:
Do you know in what situations `__linux__` will not be defined but the code still works?

Reply:
Hmm, I'm not sure I can point to where `__linux__` was introduced, but I picked it because it's what's [used in glibc](https://github.com/search?q=repo%3Abminor%2Fglibc%20__linux__&type=code) to test for Linux.

GCC and Clang both provide this define by default (along with less standard ones like `linux` and `__linux`) when compiling on Linux. I'd expect that any decent C compiler on Linux will do the same.

Reply:
I mean, even if we are not on Linux, if `<sys/random.h>` is not available then the include will fail anyway.  But yeah I think this is fine, since the syscalls are Linux-specific.